aten/src/ATen/test/cuda_half_test.cu

```diff
@@ -76,6 +76,13 @@ __device__ void test(){
   assert(::abs(::isnan(Half(0.0)) - ::isnan(0.0f)) <= threshold);
   assert(::abs(::isinf(Half(0.0)) - ::isinf(0.0f)) <= threshold);
 #endif
+
+  // test complex<32>
+  Half real = 3.0f;
```

Comment:
it's a bit sneaky to add complex32 test here. ok for this PR but ultimately it might be nice to have a separate test file if we are gonna add more complex32 tests